tmt/steps/provision/__init__.py

```diff
@@ -1197,6 +1205,14 @@ def get_data_class(cls) -> type[GuestData]:
 
     hardware: Optional[tmt.hardware.Hardware]
 
+    environment: tmt.utils.Environment = field(
```

Comment:
At this place, a declaration with type should be enough: the value would be propagated from step data, i.e. there will be no need for `normalize` or any other callback, not even the default value.